ProfilePage.js

```diff
@@ -15,34 +15,73 @@ export class ProfilePage extends Component {
     this.props.navigation.navigate('ProfilePage', {user})
   }
 
+  contactInfo = (user) => {
+    return (
+      <View>
+        <Text style={ styles.connections }>Contact Information</Text>
+        <View style={ styles.connectionsContainer }>
+          <Text>{ user.phone_number }</Text>
+          <Text>{ user.email }</Text>
+        </View>
+      </View>
+    )
+  }
+
+  editProfile = () => {
+    this.props.navigation.navigate('EditProfile')
+  }
+
+  displayEditProfile = (user) => {
+    return (
+      <Icon name='edit' size={20} color='#4AA9C5' style={ styles.editIcon } onPress={ this.editProfile } />
+    )
+  }
+
   render() {
     const user = this.props.navigation.getParam('user') ? this.props.navigation.getParam('user') : this.props.user
-    
+    const isConnection = this.props.user.connections.filter(connection => parseInt(user.id) === parseInt(connection.id))
+    const connectionInfo = isConnection.length ? this.contactInfo(user) : null
+    const editProfileSection = this.props.user.id !== user.id ? null : this.displayEditProfile(user)
+
     return (
       <ScrollView style={ styles.scrollContainer }>
         <View style={ styles.container }>
           <View style={ styles.imageContainer }>
             <Image style={ styles.profilePicture } source={ require('./profile-pic.jpeg') } />
+            {
+              editProfileSection
+            }
           </View>
-            <View style={styles.profileContainer}>
+          <View style={styles.profileContainer}>
             <View style={ styles.about }>
               <Text style={ styles.name }>{ user.name }</Text>
               <Text style={ styles.position }>{ user.position.job_title }</Text>
               <Text style={ styles.company }>{ user.employer.name }</Text>
-              <View style={ styles.locationContainer}>
+              <View style={ styles.iconContainer }>
                 <Icon name='map-pin' size={20} color='#4AA9C5' style={{marginRight: 7}} />
                 <Text style={ styles.location }>{ user.location.city }</Text>
               </View>
+              <View style={ styles.iconContainer }> 
+                <Icon name='linkedin' size={ 20 } color='#4AA9C5' style={ {marginRight: 5} }/>
+                <Text>{ user.linkedin }</Text>
+              </View>
+              <View style={ styles.iconContainer }>
+                <Icon name='github' size={ 20 } color='#4AA9C5' style={ {marginRight: 5} }/>
+                <Text>{ user.github }</Text>
+              </View>
               <Text style={ styles.bio }>{ user.bio }</Text>
               {
                 this.props.user !== user && <TouchableHighlight style={styles.connectBtn}><Button title='Connect' color='white' /></TouchableHighlight>
```

Comment:
Could we set this as a variable and put it at the top with the other ternaries to add to the consistency?

Reply:
I took the functionality out of the return 😄